mergify_cli/ci_issues/utils.py

```diff
@@ -0,0 +1,47 @@
+import pathlib
+
+import httpx
+
+from mergify_cli import console
+from mergify_cli import utils
+
+
+async def raise_for_status(response: httpx.Response) -> None:
+    if response.is_error:
+        await response.aread()
+        details = response.text or "<empty_response>"
+        console.log(f"[red]Error details: {details}[/]")
+
+    response.raise_for_status()
+
+
+def get_ci_issues_client(
+    api_url: str,
+    token: str,
+) -> httpx.AsyncClient:
+    return utils.get_http_client(
+        api_url,
+        headers={
+            "Authorization": f"Bearer {token}",
+        },
+        event_hooks={
+            "request": [],
+            "response": [raise_for_status],
+        },
+    )
+
+
+def get_files_to_upload(
+    files: tuple[str, ...],
+) -> list[tuple[str, tuple[str, bytes, str]]]:
+    files_to_upload = []
+
+    for file in set(files):
+        file_path = pathlib.Path(file)
+        # TODO(leo): stream the files instead of loading them
+        with file_path.open("rb") as f:
+            files_to_upload.append(
+                ("files", (file_path.name, f.read(), "application/xml")),
```

Comment:
Correct me if I'm wrong but it does not seem like you do not have to read the file content explicitly but you can just pass a file object to httpx.
That should prevent loading a lot of things in memory for nothing.

Reply:
Yes you're right, works fine passing the file object